docs/streaming.md

```diff
@@ -0,0 +1,95 @@
+# Spark Streaming
+
+Spark Streaming is an extension of the core Spark API that enables scalable, high-throughput, fault-tolerant stream processing of live data streams. Data can be ingested from many sources like Kafka, Flume, Kinesis, or TCP sockets, and can be processed using complex algorithms expressed with high-level functions like map, reduce, join and window. 
+
+### The Basic Idea
+
+#### Spark Streaming
+Here is a basic Spark Streaming sample which writes to the console with `wordCounts.print()`:
+
+Create a StreamingContext with a SparkConf configuration
+```scala
+    val ssc = new StreamingContext(sparkConf, Seconds(1))
+```
+
+Create a DStream that will connect to serverIP:serverPort 
+```scala
+    val lines = ssc.socketTextStream(serverIP, serverPort)
+```
+
+Count each word in each batch
+```scala
+    val words = lines.flatMap(_.split(" "))
+    val pairs = words.map(word => (word, 1))
```

Comment:
as per @mdigan's request, you can just add:
val pairs = words.filter(word => word != "ERROR'').map(word => (word, 1))